src/NuGet.Core/NuGet.Packaging/TopologicalSortUtility.cs

```diff
@@ -38,19 +38,25 @@ public static class TopologicalSortUtility
 
         private static int GetParentCount(List<PackageDependencyInfo> packages, string id)
         {
-            int count = 0;
+            var parentCount = 0;
+            var count = packages.Count;
 
-            foreach (var package in packages)
+            for (var i = 0; i < count; i++)
```

Comment:
`foreach` on a List is ugly; the enumerator is quite large and keeps a version to watch for changes